src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -413,11 +424,15 @@ private void DoStart()
             if (GlobalConfig.FastKillOnCancelKeyPress)
                 Console.CancelKeyPress += HandleProcessExit;
 
+            //TODO: setup thead pool directly to lifecycle
             ConfigureThreadPoolAndServicePointSettings();
 
-            // This has to start first so that the directory system target factory gets loaded before we start the router.
-            grainTypeManager.Start();
-            runtimeClient.Start();
+            return Task.CompletedTask;
+        }
+
+        private Task OnRuntimeServicesStart(CancellationToken ct)
+        {
```

Comment:
BTW, you can do an `await Task.Yield();` to make sure the method releases immediately instead of blocking synchronously and can run in parallel with others.
You can also address the trivial WaitWithThrow calls and replace with await WithTimeout, but as we mentioned in the other comment, I wouldn't convert everything to async at this point. It will be a little bit cleaner (in terms of how exceptions propagate), but the reality is that making them truly async is not a boost in perf or anything like that (as long as at least the method yields fast when starting the lifecycle method)